custom/icds/messaging/indicators.py

```diff
@@ -237,20 +260,21 @@ class AWWSubmissionPerformanceIndicator(AWWIndicator):
     def __init__(self, domain, user):
         super(AWWSubmissionPerformanceIndicator, self).__init__(domain, user)
 
-        self.last_submission_date = ApplicationStatusFact.objects.filter(
-            user_dim__user_id=user.get_id,
-            domain=domain,
-        ).aggregate(value=Max("last_form_submission_date"))["value"]
+        result = AggregateInactiveAWW.objects.filter(
+            awc_site_code=user.raw_username,
+            last_submission__isnull=False
+        )
+        if result:
```

Comment:
There's are a few inefficiencies with this query. 

A) I think that checking the result like this results in the full query being run without a limit. It appears that this code only cares about the first result, so it should look similar to the following so a LIMIT 1 is included:


```python
result = AggregateInactiveAWW.objects.filter(...).first()

if result: ...
```

B) If you only care about one value, you can use `values('last_submission')` so that you don't fetch the entire row. 

C) I think in practice there will only be one row returned because awc_site_code should be as unique as awc_id, but awc_id is the primary key on this table and the index added for awc_site_code is not unique.

All that to say, I think if I were to re-write this query I think I would do something similar to:

```python
result = AggregateInactiveAWW.objects.filter(awc_id=user.commcare_location_id).values('last_submission').first()
if result:
    self.last_submission_date = result
```

(Note: I haven't run that and you may need to change some of the syntax or location_id attribute)

Reply:
Thanks. https://github.com/dimagi/commcare-hq/pull/25690